### PR TITLE
fix #473 but not sure…

### LINKED ISSUE #473: Relationship fields not work with custom ownerKey
(A) what you did
Open update form with select (1-n relationship) / select2 (1-n relationship)
with relation belongsTo() with NOT default OwnerKey (default is id )

(B) what you expected to happen
I want right values for options in select input described in belongsTo() method

(C) what happened
Then i see wrong values in <option value="">
Then iam press save and got Exception, because update() try to set wrong walue (iam have foreign keys)

(D) what have you already tried to fix it?

im wrote fix, and maybe iam make pull request
here is problem : `$connected_entity_entry->getKey()`
`getKey()` method returns `id` but in custom ownerKey this value different

### PR DESCRIPTION
**Pull request by errogaht**
… about select2_multiple.blade.php and select_multiple.blade.php (please test it)

https://github.com/Laravel-Backpack/CRUD/issues/473

### CODE REVIEW

File: src/resources/views/fields/select.blade.php [hunk 1]
```diff
@@ -6,6 +6,7 @@
     @include('crud::inc.field_translatable_icon')
 
     <?php $entity_model = $crud->model; ?>
```
Comment (lloy0076) [inline src/resources/views/fields/select.blade.php]: The below line should be merged with this.

File: src/resources/views/fields/select.blade.php [hunk 2]
```diff
@@ -17,9 +18,10 @@
 
             @if (isset($field['model']))
                 @foreach ($field['model']::all() as $connected_entity_entry)
-                    <option value="{{ $connected_entity_entry->getKey() }}"
+                    <?php $value = $owner_key === null ? $connected_entity_entry->getKey() : $connected_entity_entry->{$owner_key} ?>
```
Comment (lloy0076) [inline src/resources/views/fields/select.blade.php]: Consider `is_null($owner_key)` maybe? I get antsy about a comparison to `null` itself.

File: src/resources/views/fields/select2.blade.php [hunk 3]
```diff
@@ -3,6 +3,7 @@
     <label>{!! $field['label'] !!}</label>
     @include('crud::inc.field_translatable_icon')
     <?php $entity_model = $crud->model; ?>
```
Comment (lloy0076) [inline src/resources/views/fields/select2.blade.php]: The below could be merged with this.

File: src/resources/views/fields/select2.blade.php [hunk 4]
```diff
@@ -14,8 +15,9 @@
 
             @if (isset($field['model']))
                 @foreach ($field['model']::all() as $connected_entity_entry)
-                    <option value="{{ $connected_entity_entry->getKey() }}"
-                        @if ( ( old($field['name']) && old($field['name']) == $connected_entity_entry->getKey() ) || (isset($field['value']) && $connected_entity_entry->getKey()==$field['value']))
+                    <?php $value = $owner_key === null ? $connected_entity_entry->getKey() : $connected_entity_entry->{$owner_key} ?>
```
Comment (lloy0076) [inline src/resources/views/fields/select2.blade.php]: See above re: `is_null`.

File: src/resources/views/fields/select2_multiple.blade.php [hunk 5]
```diff
@@ -2,15 +2,19 @@
 <div @include('crud::inc.field_wrapper_attributes') >
     <label>{!! $field['label'] !!}</label>
     @include('crud::inc.field_translatable_icon')
+    <?php $entity_model = $crud->model; ?>
```
Comment (lloy0076) [inline src/resources/views/fields/select2_multiple.blade.php]: And the below should be merged.

File: src/resources/views/fields/select2_multiple.blade.php [hunk 6]
```diff
@@ -2,15 +2,19 @@
...
     <select
         name="{{ $field['name'] }}[]"
         @include('crud::inc.field_attributes', ['default_class' =>  'form-control select2'])
         multiple>
 
         @if (isset($field['model']))
             @foreach ($field['model']::all() as $connected_entity_entry)
-                <option value="{{ $connected_entity_entry->getKey() }}"
-                    @if ( (isset($field['value']) && in_array($connected_entity_entry->getKey(), $field['value']->pluck($connected_entity_entry->getKeyName(), $connected_entity_entry->getKeyName())->toArray())) || ( old( $field["name"] ) && in_array($connected_entity_entry->getKey(), old( $field["name"])) ) )
+                <?php $primaryKeyName = $owner_key === null ? $connected_entity_entry->getKeyName() : $owner_key ?>
```
Comment (lloy0076) [inline src/resources/views/fields/select2_multiple.blade.php]: `is_null`.

File: src/resources/views/fields/select_multiple.blade.php [hunk 7]
```diff
@@ -2,6 +2,8 @@
 <div @include('crud::inc.field_wrapper_attributes') >
     <label>{!! $field['label'] !!}</label>
     @include('crud::inc.field_translatable_icon')
+    <?php $entity_model = $crud->model; ?>
```
Comment (lloy0076) [inline src/resources/views/fields/select_multiple.blade.php]: Merge with below.

File: src/resources/views/fields/select_multiple.blade.php [hunk 8]
```diff
@@ -12,8 +14,10 @@ class="form-control"
 
     	@if (isset($field['model']))
     		@foreach ($field['model']::all() as $connected_entity_entry)
-    			<option value="{{ $connected_entity_entry->getKey() }}"
-					@if ( (isset($field['value']) && in_array($connected_entity_entry->getKey(), $field['value']->pluck($connected_entity_entry->getKeyName(), $connected_entity_entry->getKeyName())->toArray())) || ( old( $field["name"] ) && in_array($connected_entity_entry->getKey(), old( $field["name"])) ) )
+                <?php $primaryKeyName = $owner_key === null ? $connected_entity_entry->getKeyName() : $owner_key ?>
```
Comment (lloy0076) [inline src/resources/views/fields/select_multiple.blade.php]: `is_null`.

**Comment by tabacitu** [comment]
Issue got closed. Closing PR too.

**Pull request closed without merging**